app/src/main/java/org/mozilla/focus/widget/BrowserNavigationOverlay.kt

```diff
@@ -127,6 +128,17 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         navButtonForward.isFocusable = canGoForward
 
         navUrlInput.setText(navigationProvider?.getCurrentUrl())
+
+        updateFocusOnNavigation()
+    }
+
+    private fun updateFocusOnNavigation() {
+        when {
+            findFocus() != null -> return
+            navButtonBack.isEnabled -> navButtonBack.requestFocus()
```

Comment:
I think we should requestFocus on the parent of these views, which can use `android:descendantFocusability` to ensure it's children are focused first, rather than writing out the focus patterns explicitly in code. So basically, "if visibility changes so view is shown, requestFocus on parent, which delegates to first focusable child".